icevision/models/torchvision_models/faster_rcnn/show_results.py

```diff
@@ -31,3 +37,120 @@ def show_results(
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+
+    Returns
+    -------
+    samples_plus_losses: those are the samples (dictionaries) contained in the original dataset
+                          enriched with losses.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+
+    device = model_device(model)
+    dl = valid_dl(dataset, batch_size=1, num_workers=0, shuffle=False)
+
+    samples_plus_losses = []
+    losses_stats = {
+        "loss_classifier": [],
+        "loss_box_reg": [],
+        "loss_objectness": [],
+        "loss_rpn_box_reg": [],
+        "loss_total": [],
+    }
+
+    with torch.no_grad():
+        for (x, y), sample in pbar(dl):
+            torch.manual_seed(0)
+            x = [x[0].to(device)]
+            loss = model(x, y)
+            loss = {k: float(v.cpu().numpy()) for k, v in loss.items()}
+            loss["loss_total"] = sum(loss.values())
+
+            for l in losses_stats.keys():
+                losses_stats[l].append(loss[l])
+
+            sample[0].update(loss)
+            samples_plus_losses.append(sample[0])
+
+    losses_stats = {k: get_stats(v) for k, v in losses_stats.items()}
+    return samples_plus_losses, losses_stats
+
+
+def add_annotations(samples: List[dict]) -> List[dict]:
+    """
+    Adds a `text` field to the sample dict to use as annotations when plotting.
+    """
+    for sample in samples:
+        text = ""
+        for key in sample.keys():
+            if "loss" in key:
+                text += f"{key}: {round(sample[key], 5)}\n"
+        text += f"IMG: {sample['filepath'].name}"
+        sample["text"] = text
+    return samples
+
+
+def plot_top_losses(
+    model: nn.Module,
+    dataset: Dataset,
+    sort_by: str = "loss_total",
+    n_samples: int = 5,
+) -> Tuple[List[dict], List[dict], dict]:
+    """
+    Gets a dataset and a model as input. Calculates losses for each sample in the dataset.
+    Sorts samples by `sort_by` (e.g. one of the losses). Runs model predictions on samples.
+    Plots the top sorted `n_samples` and returns losses + predictions + losses_stats.
+
+    Arguments
+    ---------
+    model: nn.Module
+    dataset: Dataset
+    sort_by: (str) the loss to sort samples by
+    n_samples: how many samples to show
+
+    Returns
+    -------
+    sorted_samples: those are the samples (dictionaries) contained in the original dataset
+                    enriched with losses and sorted by `sort_by`.
+    sorted_preds: `model` predictions on `sorted_samples`. Same order as `sorted_samples`.
+    losses_stats: dictionary containing losses stats (min, max, mean, 1-25-50-75-99 quantiles)
+                  for each one of the losses.
+    """
+    samples, losses_stats = get_losses(model, dataset)
+    samples = add_annotations(samples)
+
+    dl = infer_dl(dataset, batch_size=16)  # setting `batch_size` to 16 is arbitrary.
```

Comment:
Should we have `batch_size` as a function parameter? The only problem that I see is that `16` might be too big of a batch size for some high resolution datasets and memory might overflow

Reply:
Yes indeed. I had forgotten to do that actually :D

Reply:
Done